server/src/main/java/io/druid/server/coordination/ZkCoordinator.java

```diff
@@ -256,8 +256,14 @@ public void loadLocalCache()
       File file = segmentsToLoad[i];
       log.info("Loading segment cache file [%d/%d][%s].", i, segmentsToLoad.length, file);
       try {
-        DataSegment segment = jsonMapper.readValue(file, DataSegment.class);
-        if (serverManager.isSegmentCached(segment)) {
+        final DataSegment segment = jsonMapper.readValue(file, DataSegment.class);
+
+        if (!segment.getIdentifier().equals(file.getName())) {
```

Comment:
Should we be deleting the file if there's a mismatch instead of just ignoring it?